sound/soc/soc-pcm.c

```diff
@@ -1913,6 +1913,17 @@ int dpcm_be_dai_hw_params(struct snd_soc_pcm_runtime *fe, int stream)
 		if (!snd_soc_dpcm_can_be_params(fe, be, stream))
 			continue;
 
+		if (be->dpcm[stream].users > 1 &&
+		    be->dpcm[stream].state == SND_SOC_DPCM_STATE_HW_PARAMS) {
```

Comment:
what happens if the BE is already triggered and state is START? Should we allow hw_params then?

Reply:
it's already handled by the case above at line 1912, isn't it?

Reply:
ahh yes. sorry it wasnt easy to tell from the patch :).